.github/workflows/ci-build.yml

```diff
@@ -131,12 +144,22 @@ jobs:
         unzip opengfx-all.zip
         rm -f opengfx-all.zip
 
-    - name: CMake
-      run: mkdir build && cd build && cmake ..
+    - name: CMake - x86_64
+      run: mkdir build-x86_64 && cd build-x86_64 && cmake .. -DCMAKE_OSX_ARCHITECTURES=x86_64 -DVCPKG_TARGET_TRIPLET=x64-osx -DCMAKE_TOOLCHAIN_FILE=/usr/local/share/vcpkg/scripts/buildsystems/vcpkg.cmake
 
     - uses: ammaraskar/gcc-problem-matcher@master
-    - name: Build
-      run: cd build && make -j2
+    - name: Build - x86_64
+      run: cd build-x86_64 && make -j2
 
-    - name: Test
-      run: cd build && make -j2 test
+    - name: Test - x86_64
+      run: cd build-x86_64 && make -j2 test
+
+    - name: CMake - arm64
+      run: mkdir build-arm64 && cd build-arm64 && cmake .. -DCMAKE_OSX_ARCHITECTURES=arm64 -DVCPKG_TARGET_TRIPLET=arm64-osx -DCMAKE_TOOLCHAIN_FILE=/usr/local/share/vcpkg/scripts/buildsystems/vcpkg.cmake -DHOST_BINARY_DIR=../build-x86_64
```

Comment:
As this is just testing, and Mac OS is known for taking its sweet time to compile .. should this be 2 steps? Of course you need to do a quick `make tools` on x86-64 to get `strgen` and co, but it might speed up the CI drasticly?

CI used to be done in ~10 minutes .. with this change, that is now ~22 minutes :P Hence the question :)

You can use `matrix` if you want to make this all pretty and without less code duplication, if you like :)

Reply:
We can't run `make test` on the CI anyway unless/until they introduce arm64 hardware. :) I was using the CI as a testbed but I suspect we will either keep the CI as it is for Mac (perhaps switching to vcpkg for consistency), or add arm64 as a separate job so it can run simultaneously.

It's the release pipeline I'm intending to target for arm64 primarily. Same for Windows at some point too.

Reply:
For the release build, I also suggest splitting it up in 2 steps, for exactly the same reason ;) I was more wondering if you still want to combine the binaries or leave it at that? If you want to combine, 2 steps is still possible btw, but would require some more work. If you want to leave it as 2 binaries (you know I prefer that :P), making 2 steps is easy, but might be worth trying out while at it :)

Reply:
From a user experience point of view, a Universal binary would be preferable, but we can always try it out and see if it causes a major issue in terms of size. When I get a chance to look at this further I'll see if I can set it up with two separate steps. Building as universal basically involves running one command to glue together the two generated `openttd` binaries, before running cpack.